src/interpreter_funcs/import_obj_mesh.rs

```diff
@@ -82,18 +102,54 @@ impl<C: ObjCache> Func for FuncImportObjMesh<C> {
         log: &mut dyn FnMut(LogMessage),
     ) -> Result<Value, FuncError> {
         let path = args[0].unwrap_string();
-        let analyze = args[1].unwrap_boolean();
+        let move_to_origin = args[1].unwrap_boolean();
+        let snap_to_ground = args[2].unwrap_boolean();
+        let analyze = args[3].unwrap_boolean();
 
         let result = self.importer.import_obj(path);
         match result {
             Ok(models) => {
                 if models.is_empty() {
                     Err(FuncError::new(FuncImportObjMeshError::Empty))
                 } else {
-                    let meshes: Vec<_> = models
-                        .into_iter()
-                        .map(|model| Arc::new(model.mesh))
-                        .collect();
+                    let meshes: Vec<_> = if move_to_origin || snap_to_ground {
+                        let meshes_iter = models.into_iter().map(|model| model.mesh);
+
+                        let bboxes = meshes_iter.clone().map(|mesh| mesh.bounding_box());
+                        let union_box = BoundingBox::union(bboxes).expect("No valid meshes");
+
+                        let translation_vector = if move_to_origin {
+                            Point3::origin() - union_box.center()
+                        } else {
+                            Vector3::zeros()
+                        } + if snap_to_ground {
+                            Vector3::new(0.0, 0.0, union_box.diagonal().z / 2.0)
```

Comment:
- `move_to_origin` + `snap_to_ground` work
- standalone `move_to_origin` works
- standalone `snap_to_ground` does not work

I think we need to do different work for `snap_to_ground` depending on the state of `move_to_origin`. I suggest a match statement similar to:

```rust
let translation_vector = match (move_to_origin, snap_to_ground) {
    (true, true) => { /*...*/ }
    (true, false) => { /*...*/ }
    (false, true) => { /*...*/ }
    (false, false) => { /*...*/ }
}
```

You will need to apply the same fix to the other func.

Reply:
Are you sure? I doublechecked the code and tested the behavior and it seems to works well.

Reply:
Unless it behaves different on my machine, I am pretty sure. I tested with `bandaska.obj` (and others), both import and the snap to ground.

Reply:
Of course! I used `tatra` for testing and that one coincidentally works.